src/main/java/io/nats/client/ErrorListener.java

```diff
@@ -129,4 +130,46 @@ enum FlowControlSource { FLOW_CONTROL, HEARTBEAT }
      * @param source enum indicating flow control handling in response to which type of message
      */
     default void flowControlProcessed(Connection conn, JetStreamSubscription sub, String subject, FlowControlSource source) {}
+
+    /**
+     * Called by the connection when a low level socket write timeout occurs.
+     *
+     * @param conn The connection that had the issue
+     */
+    default void socketWriteTimeout(Connection conn) {}
+
+    /**
+     * General message producing function which understands the possible parameters to listener calls.
+     * @param label the label for the message
+     * @param conn The connection that had the issue, if provided.
+     * @param consumer The consumer that is being marked slow, if applicable
+     * @param sub the JetStreamSubscription that this occurred on, if applicable
+     * @param pairs custom string pairs. I.E. "foo: ", fooObject, "bar-", barObject will be appended
+     *              to the message like ", foo: &lt;fooValue&gt;, bar-&lt;barValue&gt;".
+     * @return
+     */
+    default String supplyMessage(String label, Connection conn, Consumer consumer, Subscription sub, Object... pairs) {

```

Comment:
This function used to be part of the ErrorListenerLoggerImpl class, but it was moved down into the interface since so it can be more easily used in ErrorListener implementations.